cpp/src/traversal/two_hop_neighbors.cuh

```diff
@@ -25,111 +25,106 @@
 #define MAXBLOCKS 65535
 #define TWO_HOP_BLOCK_SIZE 512
 
-template<typename IndexType>
+template<typename edge_t>
 struct degree_iterator {
-	IndexType* offsets;
-	degree_iterator(IndexType* _offsets) :
-			offsets(_offsets) {
-	}
-
-	__host__    __device__
-	IndexType operator[](IndexType place) {
-		return offsets[place + 1] - offsets[place];
-	}
+  edge_t const * offsets;
+  degree_iterator(edge_t const* _offsets): offsets(_offsets) {
+  }
+
+  __host__ __device__ edge_t operator[](edge_t place) {
+    return offsets[place + 1] - offsets[place];
+  }
 };
 
-template<typename It, typename IndexType>
+template<typename It, typename edge_t>
 struct deref_functor {
-	It iterator;
-	deref_functor(It it) :
-			iterator(it) {
-	}
-
-	__host__    __device__
-	IndexType operator()(IndexType in) {
-		return iterator[in];
-	}
+  It iterator;
+  deref_functor(It it): iterator(it) {
+  }
+
+  __host__ __device__ edge_t operator()(edge_t in) {
+    return iterator[in];
+  }
 };
 
 template<typename IndexType>
 struct self_loop_flagger {
-	__host__ __device__
-	bool operator()(const thrust::tuple<IndexType, IndexType> pair) {
-		if (thrust::get<0>(pair) == thrust::get<1>(pair))
-			return false;
-		return true;
-	}
+  __host__ __device__
+  bool operator()(const thrust::tuple<IndexType, IndexType> pair) {
+    if (thrust::get<0>(pair) == thrust::get<1>(pair))
+      return false;
+    return true;
+  }
 };
 
 template<typename IndexType>
 __device__ IndexType binsearch_maxle(const IndexType *vec,
-																			const IndexType val,
-																			IndexType low,
-																			IndexType high) {
-	while (true) {
-		if (low == high)
-			return low; //we know it exists
-		if ((low + 1) == high)
-			return (vec[high] <= val) ? high : low;
-
-		IndexType mid = low + (high - low) / 2;
-
-		if (vec[mid] > val)
-			high = mid - 1;
-		else
-			low = mid;
-	}
+                                     const IndexType val,
```

Comment:
typename consistency would be great. I found IndexType ambiguous now that we have VT and ET.

Reply:
I can change this.  I didn't tease through the entire implementation to understand which things where VT and which things were ET, I just let the compiler figure it out for me.  I'll work through the logic and update them.